src/data.ts

```diff
@@ -17,16 +17,33 @@ export interface DataFormatBase {
 
 export interface CsvDataFormat extends DataFormatBase {
   /**
-   * Type of input data: `"json"`, `"csv"`, `"tsv"`.
+   * Type of input data: `"json"`, `"csv"`, `"tsv"`, `"dsv"`.
    * The default format type is determined by the extension of the file URL.
    * If no extension is detected, `"json"` will be used by default.
    */
   type?: 'csv' | 'tsv';
 }
 
+export interface DsvDataFormat extends DataFormatBase {
+  /**
+   * Type of input data: `"json"`, `"csv"`, `"tsv"`, `"dsv"`.
+   * The default format type is determined by the extension of the file URL.
+   * If no extension is detected, `"json"` will be used by default.
+   */
+  type?: 'dsv';
```

Comment:
Better move `type` to DataFormatBase and make it generic so there is only single source of description

Reply:
I tried this before but it didn't work with the schema generator.